14/solve_both.rb

```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/env ruby
+lines = File.readlines(ARGV[0]).map(&:strip)
+total_seconds = 2503
+
+# Base class for reindeers
+class Reindeer
+  attr_reader :distance
+  attr_accessor :points
+
+  def initialize(line)
+    matches = line.match(/(\w+) can fly (\d+) .*for (\d+) .*for (\d+)/)
+    @name = matches[1]
+    @speed = matches[2].to_i
+    @max_running_time = matches[3].to_i
+    @resting_time = matches[4].to_i
+    @running = true
+    @time_counter = 0
+    @distance = 0
+    @points = 0
+  end
+
+  def next_sec
+    @time_counter += 1
+    if @running
```

Comment:
Because there is now ternary operator in PowerShell and I taught myself not using it :)